tests/test_context.py

```diff
@@ -39,3 +39,22 @@ def test_cpu_info_provider() -> None:
     assert res["frequency"] > 0
     assert res["num_cpus"] > 0
     assert res["total_memory"] > 0
+
+
+def test_flatten_nested_dictionary():
+    nested_ctx = Context({"a": 1, "b": {"c": 2, "d": {"e": 3}}})
+    flattened = nested_ctx.flatten(inplace=False)
+    assert flattened._ctx_dict == {"a": 1, "b.c": 2, "b.d.e": 3}
```

Comment:
I'd love to see some `keys|values|items` tests, specifically for their typing and to see that they behave expectedly.